cc/sgx/services/fns/sgx_trusted_lib/src/trusted_worker/operators.rs

```diff
@@ -156,119 +154,147 @@ impl Worker for OperatorWorker {
-                let operand_1 = b1.to_bytes_be();
-                let op2 = BigUint::parse_bytes(input.operand_2.as_bytes(), 10)
-                    .ok_or_else(|| Error::from(ErrorKind::InvalidInputError))?;
-
-                let operand_1 =
-                    eigen_crypto::ec::suite_b::ecies::aes_decrypt_less_safe(aes_key, &operand_1)
-                        .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
-
-                // TODO: Second, do ECIES decrypt
+                // 1. Cipher is encoded as base64, should be decoded
+                let cipher_operand_1 = base64::decode(&input.operand_1)
+                    .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
 
-                let op1 = operand_1;
+                // 2. Do ECIES decrypt
+                let key_pair = register_func::get_key_pair();
+                let s1 = vec![];
+                let s2 = vec![];
+                let op1 = eigen_crypto::ec::suite_b::ecies::decrypt(
+                    key_pair,
+                    &cipher_operand_1,
+                    &s1,
+                    &s2,
+                )
+                .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
                 let op1 = BigUint::from_bytes_be(&op1[..]);
 
+                // 3. Do calculate
+                let op2 = BigUint::parse_bytes(input.operand_2.as_bytes(), 10)
+                    .ok_or_else(|| Error::from(ErrorKind::InvalidInputError))?;
+
                 let result = match input.op {
                     OperatorKind::AddCipherPlain => op1 + op2,
                     OperatorKind::SubCipherPlain => op1 - op2,
                     _ => return Err(Error::from(ErrorKind::InvalidInputError)),
                 };
 
-                // TODO: First, do ECIES encrypt
-
-                let cipher = result.to_bytes_be();
-
-                // Second, do AES encrypt
-                let aes_key = register_func::get_aes_key();
-                let cipher =
-                    eigen_crypto::ec::suite_b::ecies::aes_encrypt_less_safe(aes_key, &cipher)
-                        .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
-
-                let b = BigUint::from_bytes_be(&cipher[..]);
-                let result = b.to_str_radix(10);
+                // 4. Do ECIES encrypt
+                let s1 = vec![];
+                let s2 = vec![];
+                let key_pair = register_func::get_key_pair();
+                let alg = &eigen_crypto::sign::ecdsa::ECDSA_P256_SHA256_ASN1;
+                let public_key =
+                    eigen_crypto::sign::ecdsa::UnparsedPublicKey::new(alg, key_pair.public_key());
+                let cipher = eigen_crypto::ec::suite_b::ecies::encrypt(
+                    &public_key,
+                    &s1,
+                    &s2,
+                    &result.to_bytes_be(),
+                )
+                .map_err(|_| Error::from(ErrorKind::InvalidInputError))?;
```

Comment:
Does it work to return   encrypt(xxxx )?,   https://github.com/ieigen/eigen-crypto/blob/merge/src/errors.rs   this is the error definition of crypto, and can be registered in https://github.com/ieigen/ieigen/blob/main/cc/sgx/eigen_core/src/error.rs#L420, as thus you can return by encrypt(xxxx )?,

Reply:
Indeed, it's better to return `CryptoError` instead of `InvalidInputError` here.
And is great to implement a `From` trait to make it easily to use without "ugly" `map_or`. However, it should import dependency `eigen_crypto` in `eigen_core`. Is it OK?

Reply:
I update the code with a brute `map_or` for the latest commit `f0dad68`.